Project2/Part_B/Deck.cpp

```diff
@@ -59,7 +59,28 @@ void Deck::shuffle(void)
   }
 }
 
-std::ostream& operator<<(std::ostream& os, Deck rhs)
+Card* Deck::deal(void)
+{
+  Card* dealt = this->header;
+  this->header = dealt->getNext();
+  dealt->setNext(nullptr);
+  return dealt;
+}
+
+void Deck::replace(Card* card)
+// this is a bit ugly but worked best for now while the copy constructor and assignment operator
+// are both in a different branch
+{
+  Card* head = this->header;
+  while (head->getNext() != nullptr) {
+    head = head->getNext();
+  }
+  head->setNext(card);
+  card->setNext(nullptr);
```

Comment:
this effects the values of the `Card` passed to the function, it should probably be a duplicate (see https://github.com/montymi/2560-projects/compare/main...ws-replace) - or use a new Card* that you set to be copied from the passed card, which is why I didn't put a PR up for that